cosmoz-data-nav.js

```diff
@@ -435,6 +441,32 @@
 			}
 		}
 
+		_onItemsChange(change) {
+			if (!change) {
+				return;
+			}
+
+			let affectsRenderedElements = false,
+				newSelected = this.selected;
+
+			change.indexSplices.forEach(splice => {
+				affectsRenderedElements = splice.index <= this.selected + Math.floor(this.elementsBuffer / 2);
+				if (affectsRenderedElements && this.maintainSelection) {
+					newSelected = this.selected + change.indexSplices[0].addedCount;
```

Comment:
should `newSelected` be reset to `this.selected` for each splice?
And shouldn't we address the current splice in the loop?

```js
change.indexSplices.forEach(splice => {
  ...
  newSelected += splice.addedCount;
  newSelected -= splice.removed.length;
});
newSelected = Math.max(..);
```

?